README.md
[![Build Status](https://travis-ci.org/erocoar/ggpol.svg?branch=master)](https://travis-ci.org/erocoar/ggpol)
[![CRAN_Release_Badge](http://www.r-pkg.org/badges/version-ago/ggpol)](https://CRAN.R-project.org/package=ggpol)
[![CRAN_Download_Badge](http://cranlogs.r-pkg.org/badges/grand-total/ggpol)](https://CRAN.R-project.org/package=ggpol)

### About
`ggpol` adds parliament diagrams and various other visualizations and convenience functions to `ggplot2`.

### Installation
`ggpol` can be installed via CRAN:

```r
install.packages("ggpol")
```

Alternatively, the development version can be installed via GitHub:

```r
if (!require(devtools)) {
    install.packages('devtools')
}
devtools::install_github('erocoar/ggpol')
```
### Selected Features
Below are two functions added by `ggpol`. For a full overview with applications, please refer to the [vignette](https://erocoar.github.io/ggpol/).


`geom_parliament` draws a parliament diagram, clustering points along an arc by parties with each point representing a single member of parliament. 

```r
bt <- data.frame(
 parties = factor(c("CDU", "CSU", "AfD", "FDP", "SPD", 
                    "Linke", "Gruene", "Fraktionslos"),
                  levels = c("CDU", "CSU", "AfD", "FDP", "SPD", 
                             "Linke", "Gruene", "Fraktionslos")),
 seats   = c(200, 46, 92, 80, 153, 69, 67, 2),
 colors  = c("black", "blue", "lightblue", "yellow", 
             "red","purple", "green", "grey"),
 stringsAsFactors = FALSE)

ggplot(bt) + 
  geom_parliament(aes(seats = seats, fill = parties), color = "black") + 
  scale_fill_manual(values = bt$colors, labels = bt$parties) +
  coord_fixed() + 
  theme_void()
```

![Parliament Diagram](https://i.imgur.com/gvsCvBH.png)

`geom_boxjitter` produces a hybrid of box- and scatterplot. 

```r
df <- data.frame(score = rgamma(150, 4, 1), 
                 gender = sample(c("M", "F"), 150, replace = TRUE), 
                genotype = factor(sample(1:3, 150, replace = TRUE)))

ggplot(df) + geom_boxjitter(aes(x = genotype, y = score, fill = gender),
                            jitter.shape = 21, jitter.color = NA, 
                            jitter.params = list(height = 0, width = 0.04),
                            outlier.color = NA, errorbar.draw = TRUE) +
  scale_fill_manual(values = c("#ecb21e", "#812e91")) +
  theme_minimal()
```

![Hybrid Boxplot](https://i.imgur.com/zlwIs14.png)

### Roadmap
If you would like to see a certain feature, please file an issue with a detailed description.


NEWS.md
# ggpol 0.0.7

## Minor changes
- updated documentation, thanks to @romagnolid
- fixed unit error when creating shared axes, thanks to @martinschmelzer

# ggpol 0.0.6

## Minor changes
- fix `jitter.seed` argument for `geom_boxjitter`
- internal changes in `facet_share` to make it compatible with new versions of `ggplot2` and `grid`

# ggpol 0.0.5

## New features
- `geom_confmat` for easy embedding of confusion matrices in `ggplot2` pipeline.

## Minor changes
- `geom_arcbar` permits setting `sep = 0` for no separation between individual parts.

# ggpol 0.0.4

## Minor changes
- `StatBoxJitter` now inherits from `StatBoxplot` rather than `Stat`, making for slimmer bars.
- `geom_boxjitter()` does not require `x` argument anymore.
-  Added import of `ggplot2::resolution()`, `ggplot2::alpha()`, `ggplot2::position_dodge2()` and `ggplot2:PositionJitter` to `geom_boxjitter()`. (#2)
-  Added import of `ggplot2::.pt` to `geom_bartext()`.
-  Added import of ``grid::unit()`, `ggplot2::zeroGrob()` and `ggplot2::render_axes()` to `facet_share()`.


# ggpol 0.0.3

## Minor changes
- `facet_share` fixed for `ggplot` version `3.0.0`.


# ggpol 0.0.2

## New features
- `geom_bartext` for otherwise overlapping bar chart labels

## Minor changes
- `facet_share` axis spacing has been fixed for vertical direction


# ggpol 0.0.1

## Major changes
- First commit

## New features
- `facet_share` for two plots sharing an axis
- `geom_arcbar` for arc bar diagrams with optional spacing
- `geom_boxjitter` for hybrid boxplots with optional errorbars
- `geom_circle` for circles with given radius
- `geom_parliament` for parliament diagrams
- `geom_tshighlight` for timeseries highlighting


R/geom_arcbar.R
#' Create an Arc-Barchart
#' 
#' An arc bar diagram that allows for spacing between the individual arc components 
#' and spans 180 degrees.
#' 
#' @section Aesthetics:
#' geom_arcbar understands the following aesthetics (required aesthetics are in bold):
#' 
#' - **shares**
#' - **r0** - inner radius
#' - **r1** - outer radius
#' - color
#' - fill
#' - linetype
#' - alpha 
#'   
#' @inheritParams ggplot2::geom_polygon
#' @inheritParams ggplot2::stat_identity
#' @param n The number of
#' @param sep Separation between the different shares, as a total proportion of pi.
#' @export
#' 
#' @examples
#' bt <- data.frame(
#'   parties = factor(c("CDU", "CSU", "AfD", "FDP", "SPD", 
#'                      "Linke", "Gruene", "Fraktionslos"),
#'                    levels = c("CDU", "CSU", "AfD", "FDP", "SPD", 
#'                               "Linke", "Gruene", "Fraktionslos")),
#'   seats   = c(200, 46, 92, 80, 153, 69, 67, 2),
#'   colors  = c("black", "blue", "lightblue", "yellow", "red",
#'               "purple", "green", "grey"),
#'   stringsAsFactors = FALSE)
#' 
#' ggplot(bt) + 
#'   geom_arcbar(aes(shares = seats, r0 = 5, r1 = 10, fill = parties)) + 
#'   scale_fill_manual(values = bt$colors) +
#'   coord_fixed() +
#'   theme_void()
geom_arcbar <- function(mapping = NULL, data = NULL, stat = "arcbar",
                         position = "identity", n = 360, sep = 0.05, na.rm = FALSE,
                         show.legend = NA, inherit.aes = TRUE, ...) {
  layer(data = data, 
        mapping = mapping, 
        stat = stat, 
        geom = GeomArcbar,
        position = position, 
        show.legend = show.legend, 
        inherit.aes = inherit.aes,
        params = list(na.rm = na.rm, n = n, sep = sep, ...))
}

#' @rdname ggpol-extensions
#' @format NULL
#' @usage NULL
#' @export
GeomArcbar <- ggproto(
  "GeomArcbar", ggplot2::GeomPolygon,
  default_aes = list(colour = "black", fill = NA, size = 0.5, linetype = 1, alpha = NA)
)

